### PR TITLE
Converted berry and PokeBlock strings to COMPOUND_STRINGs 

### LINKED ISSUE #4959: Convert single-use string variables to COMPOUND_STRINGs
### Description

From @Bassoonian in #4951 
>These are only used once, so I would personally be inclined to convert them to COMPOUND_STRINGs and remove them from strings.c and strings.h. The only reason they're lumped with the rest of the strings is for pret matching purposes and it just makes it harder to edit imo

I think this logic should be applied to all single-use strings

- [x] Berry Blender
- [x] PokeNav
- [x] Default player names
- [x] Mechadolls
- [x] PC text
- [x] Bag pockets
- [x] Berries
- [x] PokeBlocks

### Discord contact info
Frankfurter0

### PR DESCRIPTION
**Pull request by fdeblasio**
## Description
Turns berry and PokeBlock-related string variables into COMPOUND_STRINGs for easier editing

## Issue(s) that this PR fixes
Partially fixes #4959 

## **Discord contact info**
Frankfurter0

### CODE REVIEW

File: src/berry_tag_screen.c [hunk 1]
```diff
@@ -143,11 +143,11 @@ static const struct WindowTemplate sWindowTemplates[] =
...
-    gBerryFirmnessString_VerySoft,
-    gBerryFirmnessString_Soft,
-    gBerryFirmnessString_Hard,
-    gBerryFirmnessString_VeryHard,
-    gBerryFirmnessString_SuperHard
+    COMPOUND_STRING("Very soft"),
+    COMPOUND_STRING("Soft"),
+    COMPOUND_STRING("Hard"),
+    COMPOUND_STRING("Very hard"),
+    COMPOUND_STRING("Super hard")
```
Comment (AsparagusEduardo) [inline src/berry_tag_screen.c:150]: Can you add `BERRY_FIRMNESS_` indexes to this table?

Unrelated, but it seems that this table doesn't consider `BERRY_FIRMNESS_UNKNOWN`, which might cause issues for Kee and Maranga berries. She should check them.

**Review by AsparagusEduardo — DISMISSED** [review]

File: src/berry_tag_screen.c (hunk 1 above)
Comment (fdeblasio) [reply src/berry_tag_screen.c:150]: Do you mean something like
`[BERRY_FIRMNESS_VERY_SOFT] = COMPOUND_STRING("Very soft"),`?

If we add BERRY_FIRMNESS_UNKNOWN, then I think we'll need to update src/berry_tag_screen.c to change `sBerryFirmnessStrings[berry->firmness - 1]` to `sBerryFirmnessStrings[berry->firmness]` (mainly a note for me so I don't forget tomorrow)

Comment (AsparagusEduardo) [reply src/berry_tag_screen.c:150]: Makes sense to me. It'd be better than to have them underflow imo.

Comment (fdeblasio) [reply src/berry_tag_screen.c]: Done

Comment (AsparagusEduardo) [reply src/berry_tag_screen.c]: You didn't update the call to `sBerryFirmnessStrings`.

Comment (fdeblasio) [reply src/berry_tag_screen.c]: Oops. Now it's updated

File: src/pokeblock_feed.c [hunk 2]
```diff
@@ -96,6 +96,10 @@ static u8 CreatePokeblockCaseSpriteForFeeding(void);
 static u8 CreateMonSprite(struct Pokemon *);
 static void SpriteCB_ThrownPokeblock(struct Sprite *);
 
+static const u8 gText_Var1AteTheVar2[] = _("{STR_VAR_1} ate the\n{STR_VAR_2}.{PAUSE_UNTIL_PRESS}");
+static const u8 gText_Var1HappilyAteVar2[] = _("{STR_VAR_1} happily ate the\n{STR_VAR_2}.{PAUSE_UNTIL_PRESS}");
+static const u8 gText_Var1DisdainfullyAteVar2[] = _("{STR_VAR_1} disdainfully ate the\n{STR_VAR_2}.{PAUSE_UNTIL_PRESS}");
```
Comment (Bassoonian) [inline src/pokeblock_feed.c:101]: Static text should be `sText`. This goes for the other changed files too

Comment (fdeblasio) [reply src/pokeblock_feed.c]: Done

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**